static/js/biomedical/disease/data_processing_utils.ts

```diff
@@ -83,3 +86,67 @@ export function getDiseaseGeneAssociation(
   }
   return rawData;
 }
+/**
+ * Fetches the disease-symptom association data
+ * @param data
+ * @returns
+ */
+export function getDiseaseSymptomAssociation(
+  data: GraphNodes
+): DiseaseSymptomAssociationData[] {
+  // checks if the data is empty
+  if (!data) {
+    return [];
+  }
+  const rawData: DiseaseSymptomAssociationData[] = [];
+  // checks for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "diseaseOntologyID") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let symptom = null;
+      let oddsRatioValue = null;
+      // check for null or non-existent property values
+      if (_.isEmpty(node.neighbors)) {
+        continue;
+      }
+      for (const n of node.neighbors) {
+        if (n.property === "associationOddsRatio") {
+          // check for empty list and null gene values
+          if (_.isEmpty(n.nodes) || _.isEmpty(n.nodes[0].value)) {
+            continue;
+          }
+          oddsRatioValue = n.nodes[0].value;
+        } else if (n.property === "medicalSubjectHeadingID") {
+          for (const n1 of n.nodes) {
+            if (n1.neighbors === undefined || _.isEmpty(n1.value)) {
```

Comment:
I should probably clarify this in a comment. But, we ultimately want to get the descriptor name(n2) which is nested below medicalSubjectHeadingID(n). So, if n1.neighbors is empty or null, n2 would be empty as well.

Reply:
I don't think that answers my question

1. so if n2 would be empty if n1.neighbors is empty or null, why do we only check for n1.neighbors being undefined right now?
2. why are we checking n1.value when we don't use n1.value?

Reply:
Ah, I see what you mean. We can directly check the ._isEmpty condition on n2.nodes, instead imposing another layer of checks with n1.neighbors and n1.value. You're right that n1.value isn't used at all so it isn't worth checking for it. Even for n1.neighbors, now that I think about it, we might not need that additional check. What do you think?

Reply:
For n1.neighbors, it depends, can it be undefined or null? If it is possible for n1.neighbors to be undefined or null, then we should check that before looping through it (otherwise, there will be an error)

Reply:
Got it! It might be nice to check for n1.neighbors to be null/undefined because we're not confident that they'll exist for all disease cases. So, I guess removing the n1.value check is the only change we'd want?

Reply:
remove the n1.value check & add a check for n1.neighbors not being null (currently, you're only checking undefined)

Reply:
Done!